suitcase/hdf5.py

```diff
@@ -2,21 +2,32 @@
 import warnings
 import h5py
 import json
+import copy
 from metadatastore.commands import get_events_generator
 from databroker.databroker import fill_event
 from databroker.core import Header
 
 
 __version__ = "0.2.2"
 
-def export(headers, filename):
+def export(headers, filename, stream_name=None, fields_unwanted=None, timestamps_opt=True):
     """
+    Create hdf5 file to preserve the structure of databroker. Necessary copy are made at each step, so
+    headers will not be modified.
+
     Parameters
     ----------
     headers : a Header or a list of Headers
         objects retruned by the Data Broker
     filename : string
         path to a new or existing HDF5 file
+    stream_name : string, optional
+        None means save all the data from each descriptor, i.e., user can define stream_name as primary,
+        so only data with descriptor.name == primary will be saved.
+    fields_unwanted : list, optional
```

Comment:
Yeah, I see what you're saying.  I guess the problem is that the two use cases (whitelist vs blacklist) are at odds with each other.  Perhaps the best way forward is to be consistent with the databroker's implementation and `fields_unwanted` -> `fields` and use this as a whitelist field.  Then we can also provide a helper function to take a list of headers and a set of blacklist keys and then return all the whitelist keys.

```python
def blacklist_to_whitelist(headers, unwanted_fields):
    whitelist = set()
    for header in headers:
        for descriptor in header.descriptors:
            good = [key for key in descriptor.data_keys.keys()
                    if key not in unwanted_fields]
            whitelist.update(good)
    return whitelist
```

or bonus points for a massive one-line list comprehension

```python
def blacklist_to_whitelist(headers, unwanted_fields):
    return set([key for hdr in headers
                    for key in [k for desc in hdr.descriptors
                                  for k in desc.data_keys.keys()]
                if key not in unwanted_fields])
```

And then you can do something like

```python
unwanted_fields = ['a', 'b', 'c', 'd', etc...]
whitelisted_fields = suitcase.hdf.blacklist_to_whitelist(headers, unwanted_fields)
suitcase.hdf.export(headers, whitelisted_fields)
```

Let it also be noted that I don't particularly have strong feelings on whether we make this a whitelisted fields kwarg or a blacklisted fields argument as the one that goes in the `suitcase.hdf.export()` function.  My only thinking on making this kwarg a whitelisted field instead of a blacklisted field is to be consistent with what is already provided in databroker, but I could easily be convinced otherwise.  Perhaps we need a third party here to weigh in.